tests/unit_tests/utils/test_general.py

```diff
@@ -337,3 +413,258 @@ def test_get_file_age(tmp_test_directory):
     # Ensure that the function raises FileNotFoundError for a non-existent file
     with pytest.raises(FileNotFoundError):
         gen.get_file_age(tmp_test_directory / "nonexistent_file.txt")
+
+
+def test_separate_args_and_config_data() -> None:
+    # Test the function "separate_args_and_config_data"
+    expected_args = ["arg1", "arg2"]
+    kwargs = {"arg1": 1, "arg2": 2, "arg3": 3}
+    args, config_data = gen.separate_args_and_config_data(expected_args, **kwargs)
+    assert args == {"arg1": 1, "arg2": 2}
+    assert config_data == {"arg3": 3}
+
+
+def test_get_log_excerpt(tmp_test_directory) -> None:
+    log_file = tmp_test_directory / "log.txt"
+    with open(log_file, "w", encoding="utf-8") as f:
+        f.write("This is a log file.\n")
+        f.write("This is the second line of the log file.\n")
+
+    assert gen.get_log_excerpt(log_file) == (
+        "\n\nRuntime error - See below the relevant part of the log/err file.\n\n"
+        f"{log_file}\n"
+        "====================================================================\n\n"
+        "This is a log file."
+        "This is the second line of the log file.\n\n"
+        "====================================================================\n"
+    )
+
+
+def test_file_has_text(tmp_test_directory) -> None:
+    file = tmp_test_directory / "test_file_has_text.txt"
+    text = "test"
+    with open(file, "w") as f:
+        f.write(text)
+    assert gen.file_has_text(file, text)
+    assert not gen.file_has_text(file, "test2")
```

Comment:
What is the expected return for an empty file? I guess False? Worth testing?

Reply:
The function would have failed cause you cannot mmap an empty file. Generally I am of the opinion that we should allow python to raise its own error in this case. However, in this specific case I added a `try...except` to deal with that case and the case where the file does not exist (and the appropriate tests).

Reply:
ok!